pkg/insights/deploy.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package insights
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/okteto/okteto/pkg/cmd/pipeline"
+	"github.com/okteto/okteto/pkg/k8s/configmaps"
+	"github.com/okteto/okteto/pkg/okteto"
+)
+
+const (
+	// deployInsightType represents the type of the deploy event
+	deployInsightType = "deploy"
+
+	// deploySchemaVersion represents the schema version of the deploy event
+	// This version should be updated if the structure of the event changes
+	deploySchemaVersion = "1.0"
+)
+
+// deployEventJSON represents the JSON structure of a deploy event
+type deployEventJSON struct {
+	DevenvName    string      `json:"devenv_name"`
+	Repository    string      `json:"repository"`
+	Namespace     string      `json:"namespace"`
+	SchemaVersion string      `json:"schema_version"`
+	Phase         []phaseJSON `json:"phases"`
+	Success       bool        `json:"success"`
+}
+
+// phaseJSON represents the JSON structure of a phase in a deploy event
+type phaseJSON struct {
+	Name     string  `json:"name"`
+	Duration float64 `json:"duration"`
+}
+
+// TrackDeploy tracks an image build event
+func (ip *Publisher) TrackDeploy(ctx context.Context, name, namespace string, success bool) {
+	k8sClient, _, err := ip.k8sClientProvider.Provide(okteto.GetContext().Cfg)
+	if err != nil {
+		ip.ioCtrl.Logger().Infof("could not get k8s client: %s", err)
+		return
+	}
+	cfgName := pipeline.TranslatePipelineName(name)
+	cmap, err := configmaps.Get(ctx, cfgName, namespace, k8sClient)
+	if err != nil {
+		ip.ioCtrl.Logger().Infof("could not get pipeline configmap: %s", err)
+		return
+	}
+
+	val, ok := cmap.Data["phases"]
+	// If there is no phases, we don't track the event
+	if !ok {
+		ip.ioCtrl.Logger().Infof("no phases found in pipeline configmap. Skipping event tracking")
+		return
+	}
+
+	var phases []phaseJSON
+	if err := json.Unmarshal([]byte(val), &phases); err != nil {
+		ip.ioCtrl.Logger().Infof("could not unmarshal phases from cmap: %s", err)
+		return
+	}
+
+	deployEvent := &deployEventJSON{
+		DevenvName:    name,
+		Repository:    cmap.Data["repository"],
```

Comment:
We have to obfuscate this to prevent to send username/password as part of the repository information. 

Also, should we standardize how we send the repo? I mean, `git@github.com:okteto/okteto.git`, `https://github.com/okteto/okteto`, `https://github.com/okteto/okteto.git` and `ssh://git@github.com/okteto/okteto` points to the same repository but it wouldn't be easy to be grouped by repository if we send them in a different way. I think we should unify them and send them always in the same way (always https://, always ssh:// or similar)